packages/pages/src/common/src/project/structure.ts

```diff
@@ -102,6 +109,10 @@ export class ProjectStructure {
 
   sitesConfigRoot: Path;
   templatesRoot: Path;
+  domain?: string;
+  sitesConfigDomain?: Path;
```

Comment:
I think we should spend more time thinking about the names `sitesConfigDomain` and `templatesDomain`. They feel a bit un-intuitive.

Reply:
adding onto this, can't sitesConfigDomain and templateDomain be resolved from the domain? if that's so then I think we should expose functions which abstract away how to handle the root + domain logic

Reply:
the naming here is to follow the existing variable names `sitesConfigRoot` and `templatesRoot`. Alternatively, if we are fine with deviating from that, maybe `domainSitesConfigPath`?

Reply:
the two new variables can be derived from joining the root variables with domain, but since they can potentially be use in multiple places, I thought it would be good to construct the paths and store it without the need to construct it each time.

Reply:
updated the name to `scopedTemplatesPath` and `scopedsitesConfigPath`